custom_components/tahoma/switch.py

```diff
@@ -31,47 +32,66 @@ class TahomaSwitch(TahomaDevice, SwitchEntity):
     def __init__(self, tahoma_device, controller):
         """Initialize the switch."""
         super().__init__(tahoma_device, controller)
-        self._state = STATE_OFF
-        self._skip_update = False
+
+        self._state = None
 
     def update(self):
         """Update method."""
-        # Postpone the immediate state check for changes that take time.
+
         if self.should_wait():
             self.schedule_update_ha_state(True)
             return
 
-        if self._skip_update:
-            self._skip_update = False
-            return
-
         self.controller.get_states([self.tahoma_device])
 
-        _LOGGER.debug("Update %s, state: %s", self._name, self._state)
+        if CORE_ON_OFF_STATE in self.tahoma_device.active_states:
+            self.current_value = (
+                self.tahoma_device.active_states.get(CORE_ON_OFF_STATE) == "on"
```

Comment:
use STATE_ON

Reply:
`STATE_ON` is a Home Assistant const, where "on" is a state from Tahoma. What if the Home Assistant const changes, but Tahoma not? I would like to keep them separate, however maybe good to have on / off / detected etc. also in the Tahoma const?

Reply:
Fine with me to keep them separate. But I think it would be better to have those as const, and try to not use hardcoded string outside of const.py or the header of integration files.